utils/jwtUtil.js

```diff
@@ -35,35 +42,32 @@ function assignKeyValue(object, key, value) {
 } // assignKeyValue
 
 /**
- * Assigns user related information to the request header.
- * @param  {Object}  req  - The request object
- * @param  {String}  tokenName - Token name parsed from the token
- * @param  {Boolean} isAdmin  - Flag to indicate if the request was made by an
- * Admin user
+ * Assign request headers using token info
+ * @param  {Object} req - Request object
+ * @param  {Object} decodedTokenData - decoded data from token
  */
-function assignUserHeaders(req, tokenName, isAdmin) {
-  assignKeyValue(req.headers, 'UserName', req.user.name);
-  assignKeyValue(req.headers, 'TokenName', tokenName);
-  assignKeyValue(req.headers, 'ProfileName', req.user.profile.name);
-  assignKeyValue(req.headers, 'IsAdmin', isAdmin);
-  assignKeyValue(req.headers, 'IsCollector', false);
-  assignKeyValue(req.headers, 'IsBot', false);
-} // assignUserHeaders
+function assignHeaderValues(req, decodedTokenData) {
+  // username and tokenname should always be there if createToken function is
+  // used for creating token
+  if (decodedTokenData.username) {
+    assignKeyValue(req.headers, 'UserName', decodedTokenData.username);
+  }
 
-/**
- * Assigns collector related information to the request header
- * @param  {Object} req - The request object
- * @param  {String} userName  - The username parsed from the token
- * @param  {String} tokenName - The tokenname parsed from the token
- */
-function assignCollectorHeaders(req, userName, tokenName) {
-  assignKeyValue(req.headers, 'UserName', userName);
-  assignKeyValue(req.headers, 'TokenName', tokenName);
-  assignKeyValue(req.headers, 'ProfileName', '');
-  assignKeyValue(req.headers, 'IsCollector', true);
-  assignKeyValue(req.headers, 'IsAdmin', false);
-  assignKeyValue(req.headers, 'IsBot', false);
-} // assignCollectorHeaders
+  if (decodedTokenData.tokenname) {
+    assignKeyValue(req.headers, 'TokenName', decodedTokenData.tokenname);
+  }
+
+  Object.keys(headersWithDefaults).forEach((headerName) => {
+    let headerValue;
+    if (decodedTokenData.hasOwnProperty(headerName)) {
+      headerValue = decodedTokenData[headerName];
+    } else {
+      headerValue = headersWithDefaults[headerName];
+    }
```

Comment:
maybe we can do something like 
`headerValue = decodedToken[headerName] || headersWithDefaults[headerName]`  instead of the else block

Reply:
@shriramshankar I think this will not work as we expect. Consider we add a new default value in headersWithDefaults['newHeader'] = true
And then I set newHeader to false in decodedToken['newHeader'] = false
Then, I will have req.newHeader set to true (which is opposite of what we set in token in first place).
I am going to add a test for this use case.